### PR TITLE
Pivotal ID # 169314431: Project Resource Docs

### PR DESCRIPTION
**Pull request by jhoanmanuelms**
- Use swagger annotations to describe the API
- Configure the swagger to show the documentation

### CODE REVIEW

File: submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/common/config/SecurityConfig.kt [hunk 1]
```diff
@@ -44,6 +44,10 @@ class SecurityConfig(
             .and()
             .authorizeRequests()
             .antMatchers("/auth/**").permitAll()
+            .antMatchers("/v2/**").permitAll()
```
Comment (Juan-EBI) [inline submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/common/config/SecurityConfig.kt]: what are this resources starting with `v2` ?

File: submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/resources/ProjectResource.kt [hunk 2]
```diff
@@ -18,21 +22,33 @@ import org.springframework.web.bind.annotation.RequestParam
...
 
     @PostMapping(headers = ["${HttpHeaders.CONTENT_TYPE}=$MULTIPART_FORM_DATA"])
     @ResponseBody
+    @ApiOperation("Register a new project")
+    @ApiImplicitParam(name = "X-Session-Token", value = "The authentication token", required = true)
     fun submit(
-        @AuthenticationPrincipal user: SecurityUser,
+        @ApiIgnore
+        @AuthenticationPrincipal
+        user: SecurityUser,
```
Comment (Juan-EBI) [inline submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/resources/ProjectResource.kt]: I am wondering if ` .ignoredParameterTypes(AuthenticationPrincipal.class)` does not work?

Comment (jhoanmanuelms) [reply submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/resources/ProjectResource.kt]: You're right, nice catch!

File: submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/common/config/SecurityConfig.kt (hunk 1 above)
Comment (jhoanmanuelms) [reply submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/common/config/SecurityConfig.kt]: This is the path for the swagger resource generated by the annotations, which later is used by the swagger-ui plugin to show the documentation

**Pull request merged**